shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/TemporaryTerminalRewriter.kt

```diff
@@ -141,23 +141,25 @@ class TemporaryTerminalRewriter(
     fun truncatePatternsAtStop(
         stopPatterns: NearbyStaticData.StopPatterns
     ): NearbyStaticData.StopPatterns {
-        for (fullPattern in stopPatterns.patterns.flatMap { it.patterns }) {
-            if (fullPattern.typicality != RoutePattern.Typicality.Typical) continue
-            val fullPatternStopIds =
-                globalData.trips[fullPattern.representativeTripId]?.stopIds ?: continue
+        for (fullStaticPatterns in stopPatterns.patterns) {
```

Comment:
This amount of nested `for` loops is kind of a yellow flag for me - it's probably not worth optimizing, but I wouldn't say no to some comments for clarity.

Reply:
Added a comment.